libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/ManeuverProcessor.kt

```diff
@@ -240,13 +239,21 @@ internal object ManeuverProcessor {
         } ?: throw RuntimeException("Could not find step with index $this")
     }
 
-    private fun updateDistanceRemainingForCurrentManeuver(
-        stepToManeuvers: StepIndexToManeuvers,
+    private fun StepIndexToManeuvers.updateDistanceRemainingForCurrentManeuver(
         currentInstructionIndex: Int,
         stepDistanceRemaining: Double
     ) {
-        stepToManeuvers.maneuverList[currentInstructionIndex].stepDistance.distanceRemaining =
-            stepDistanceRemaining
+        val maneuverAtCurrentInstructionIndex = maneuverList[currentInstructionIndex]
+        val maneuverWithUpdatedDistRemaining =
+            maneuverAtCurrentInstructionIndex.mapTo(stepDistanceRemaining)
+        val updatedList = maneuverList.map { currentManeuver ->
+            if (currentManeuver == maneuverAtCurrentInstructionIndex) {
```

Comment:
Instead of doing equality checks on all the elements in search of the one that we want to mutate, why don't we leverage the index that we have already access to and replace the element?

Reply:
Done.